### ISSUE TITLE
[Bug]: label-fail (from script module) works on random reloads

### ISSUE THREAD

**Issue by prabh1601** [polybar/polybar#2650]
### Checklist

- [X] I have read the appropriate section in the [contributing guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
- [X] I believe this issue is a problem with polybar itself and not a misconfiguration on my part
- [X] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that may have already reported this problem
- [X] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.
- [X] I have followed the [debugging guide](https://github.com/polybar/polybar/wiki/Debugging-your-Config) to narrow down the problem to a minimal config.

### Steps to reproduce

I believe label-fail part is still buggy, it seems to work randomly and fails to execute randomly.

1. If you randomly restart polybar, it seems to toggle between working / not working 

### Minimal config

```dosini
[module/restock]
type = custom/script
interval = 300
exec = ~/.scripts/stock.sh
 
label = %output%
label-font = 15
format =  <label>
format-background = ${colors.red}
format-foreground = ${colors.black}
format-padding = 1
format-font = 3
 
label-fail = %output%
label-fail-font = 15
label-fail-foreground = ${colors.black}
format-fail =  <label-fail>
format-fail-background = ${colors.bg-alt}
format-fail-foreground = ${colors.dark-red}
format-fail-padding = 1
format-fail-font = 3
```

### Expected behavior

The label-fail condition doesnt execute everytime, In order to fix it I have to restart polybar randomly.

### Actual behavior

see expected behavior

### Window Manager and Version

i3-gaps

### Linux Distribution

Arch Linux

### Polybar version

```text
polybar 3.6.1

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -O3 -DNDEBUG -Wall -Wextra -Wpedantic -Wsuggest-override
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -Wall -Wextra -Wpedantic -Wsuggest-override  -Wall -Wextra -Wpedantic -Wsuggest-override
```

**Comment by patrick96** [comment]
I am not totally sure I understand what's going wrong. Does polybar sometimes display `format` instead of `format-fail` even though the script exited with a non-zero exit code?

If so, you may be suffering from the issue we discovered in #2630.

**Comment by prabh1601** [comment]
> I am not totally sure I understand what's going wrong. Does polybar sometimes display format instead of format-fail even though the script exited with a non-zero exit code?

Yep, exactly this

**Comment by indev29** [comment]
I can be assigned to this, since I'll do the fix from #2630.

**Comment by patrick96** [comment]
Fixed in #2667
